src/Nethermind/Nethermind.Synchronization/SnapSync/SnapServer.cs

```diff
@@ -0,0 +1,313 @@
+        {
+            byte[][]? requestedPath = pathSet[reqi].Group;
+            switch (requestedPath.Length)
+            {
+                case 0:
+                    return null;
+                case 1:
+                    try
+                    {
+                        byte[]? rlp = tree.GetNodeByPath(Nibbles.CompactToHexEncode(requestedPath[0]), rootHash.ToCommitment());
+                        response.Add(rlp);
+                    }
+                    catch (MissingTrieNodeException)
+                    {
+                        TrackTrieNodeException(rootHash);
+                        abort = true;
+                    }
+                    break;
+                default:
+                    try
+                    {
+                        Account? account = GetAccountByPath(tree, rootHash, requestedPath[0]);
+                        if (account is null)
+                        {
+                            break;
+                        }
+                        Hash256? storageRoot = account.StorageRoot;
+                        if (storageRoot.Bytes.SequenceEqual(Keccak.EmptyTreeHash.Bytes))
+                        {
+                            break;
+                        }
+                        StorageTree sTree = new(_store, storageRoot, _logManager);
+
+                        for (int reqStorage = 1; reqStorage < requestedPath.Length; reqStorage++)
+                        {
+                            byte[]? sRlp = sTree.GetNodeByPath(Nibbles.CompactToHexEncode(requestedPath[reqStorage]));
+                            response.Add(sRlp);
+                        }
+                    }
+                    catch (MissingTrieNodeException)
+                    {
+                        TrackTrieNodeException(rootHash);
+                        abort = true;
+                    }
+                    break;
+            }
+        }
+        return response.ToArray();
+    }
+
+    public byte[][] GetByteCodes(IReadOnlyList<ValueHash256> requestedHashes, long byteLimit, CancellationToken cancellationToken)
+    {
+        long currentByteCount = 0;
+        List<byte[]> response = new(requestedHashes.Count);
+
+        if (byteLimit > HardResponseByteLimit)
+        {
+            byteLimit = HardResponseByteLimit;
+        }
+
+        foreach (ValueHash256 codeHash in requestedHashes)
+        {
+            // break when the response size exceeds the byteLimit - it is a soft limit
+            // so not a big issue if we so over slightly.
+            if (currentByteCount > byteLimit)
+            {
+                break;
+            }
+
+            if (cancellationToken.IsCancellationRequested)
+            {
+                break;
+            }
+
+            // TODO: handle empty code in the DB itself?
+            if (codeHash.Bytes.SequenceEqual(Keccak.OfAnEmptyString.Bytes))
+            {
+                response.Add(Array.Empty<byte>());
```

Comment:
missing break?

Reply:
No, looks like we can handle skipped entries. Geth usually does not include null in their list, and just skip them, so I assume they work the same.